### PR TITLE
[docs] Limit information for object model

### PR DESCRIPTION
**Pull request by ronny-mysten**
## Description 

In its original form, the Object Model document pointed to a line of code where the ProtocolConfig struct was. Weird, but the code was added to since the link was made so it no longer pointed to the intended spot. Worse, the link never showed what the actual limits were, just the param name and type. 

These edits fix the code copy plugin to locate Rust structs so the current code can be used in the document instead of creating another soon-to-be outdated link.

Also, adds a component to get the protocol configs for each network so the values are available from docs.

The Object Model content was also updated to accommodate these features.

Result is bottom of this page: https://sui-docs-git-fork-ronny-mysten-docs-87-sui-foundation.vercel.app/concepts/object-model

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by wlmyng — APPROVED** [review]
Looks cool!

File: docs/site/src/plugins/inject-code/injectLoader.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ const addCodeInject = function (source) {
                 let structContent = [];
                 for (let struct of structs) {
                   struct = struct.trim();
-                  const structStr = `^(\\s*)*?(public )?struct \\b${struct}\\b.*?}`;
+                  const structStr = `^(\\s*)*?(pub(lic)? )?struct \\b${struct}\\b.*?}`;
```
Comment (wlmyng) [inline docs/site/src/plugins/inject-code/injectLoader.js:120]: What does this do?

File: docs/site/src/components/ProtocolConfig/index.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  const parseResult = (data) => {
+    let items = Object.entries(data);
+
+    return items.map((item) => {
+      if (item[1] === null) {
+        return item;
+      }
+      if (typeof item[1] === "object") {
+        const [k, v] = Object.entries(item[1])[0];
+        return [item[0], k, v];
```
Comment (wlmyng) [inline docs/site/src/components/ProtocolConfig/index.tsx:38]: Is object a JavaScript thing or is this part of the data returned by the jsonrpc call

**Review by wlmyng — COMMENTED** [review]
And some clarifying qs

File: docs/site/src/components/ProtocolConfig/index.tsx (hunk 2 above)
Comment (ronny-mysten) [reply docs/site/src/components/ProtocolConfig/index.tsx:38]: The data returned is a json object from the call. One of two possible entries is `{"name": { type: "value"}` or `{"name": null}`. I just double-check that the second item is in fact an object so the Object.entries doesn't freak out.

Comment (ronny-mysten) [reply docs/site/src/components/ProtocolConfig/index.tsx:38]: Sorry, at this point it's either `["name", {type: "value"}]` or `["name", null]`

**Pull request merged**